File: repos_estudiantes_ALVAREZ_segunda_entrega/CAGUAYA_APAZA_YONATHAN_TONY/Practica 2/Composicion-Agregacion/ejercicio1/ejercicio1.java
public class Ejercicio1Composicion {
    public static void main(String[] args) {
        Casa casa = new Casa("Av. Siempre Viva 742");
        casa.mostrarCasa();
    }
}

class Habitacion {
    private String nombre;
    private float tamano;

    public Habitacion(String nombre, float tamano) {
        this.nombre = nombre;
        this.tamano = tamano;
    }

    public void mostrarInfo() {
        System.out.println("Habitación: " + nombre + ", Tamaño: " + tamano + " m²");
    }
}

class Casa {
    private String direccion;
    private Habitacion[] habitaciones;

    public Casa(String direccion) {
        this.direccion = direccion;
        this.habitaciones = new Habitacion[3];
        crearHabitaciones();
    }

    private void crearHabitaciones() {
        habitaciones[0] = new Habitacion("Sala", 25.0f);
        habitaciones[1] = new Habitacion("Cocina", 15.0f);
        habitaciones[2] = new Habitacion("Dormitorio", 20.0f);
    }

    public void mostrarCasa() {
        System.out.println("Dirección: " + direccion);
        for (Habitacion h : habitaciones) {
            h.mostrarInfo();
        }
    }
}

File: repos_estudiantes_TARQUI_segunda_entrega/Mallcu_Calani_Alejandro_Gabriel/Composicion Agregacion/Ejercicio1/Ejercicio1.py
class Habitacion:
    def __init__(self, nombre, tamaño):
        self.nombre = nombre
        self.tamaño = tamaño

    def mostrar_info(self):
        return f"- {self.nombre}: {self.tamaño} m²"

    def get_nombre(self):
        return self.nombre

    def get_tamaño(self):
        return self.tamaño


class Casa:
    def __init__(self, direccion):
        self.direccion = direccion
        self.habitaciones = []

    def agregar_habitacion(self, habitacion):
        self.habitaciones.append(habitacion)

    def mostrar_casa(self):
        print(f"Dirección de la casa: {self.direccion}")
        print("Habitaciones:")
        for h in self.habitaciones:
            print(h.mostrar_info())


# b) Crear casa y agregar habitaciones
casa = Casa("Calle Falsa 123")
casa.agregar_habitacion(Habitacion("Cocina", 12))
casa.agregar_habitacion(Habitacion("Sala", 20))
casa.agregar_habitacion(Habitacion("Dormitorio", 15))

# c) Mostrar información
casa.mostrar_casa()
